File: Lucky Numbers - GFG/lucky-numbers.js
//{ Driver Code Starts
//Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let n = parseInt(readLine());
        let obj = new Solution();
        if(obj.isLucky(n)){
            console.log("1");   
        }
        else{
            console.log("0");   
        }
        
    }
}
// } Driver Code Ends


//User function Template for javascript

/**
 * @param {number} n
 * @returns {boolean}
*/

class Solution {
    check(n,counter)
    {
        // variable next_position is just for readability of 
        // the program we can remove it and use n only  
       	if(counter<=n)
       	{
			if(n%counter==0)
				return false;
			//calculate next position of input number
			//Math.floor(n) is used to get largest integer smaller than n
			n=n-Math.floor(n/counter);
			counter++;
			// make recursive call with updated counter and position
			return this.check(n, counter);
       	}      
   		else
   			return true;
        
    }
    isLucky(n)
    {
        return this.check(n,2);   
    }
}

File: Lucky Numbers - GFG/lucky-numbers.java
//{ Driver Code Starts
//Initial Template for Java


import java.util.*;
import java.io.*;
import java.lang.*;

class Driver
{
    public static void main(String args[])throws IOException
    {
        BufferedReader reader =new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(reader.readLine());//testcases
        
        while(t-- > 0)
        {
            int n = Integer.parseInt(reader.readLine());//input n
            //printing "1" if isLucky() returns true else "0"
            Solution s = new Solution();
            System.out.println(s.isLucky(n)? "1" : "0");
        }
    }
}
// } Driver Code Ends


//User function Template for Java



class Solution
{
    public static boolean check(int n,int counter)
    {
        if(counter<=n){
            if(n%counter==0)
                return false;
		    // calculate next position of input number
		    n=n-n/counter;
		    counter++;
		    // make recursive call with updated counter 
		    // and position return check(n, counter);
		    return check(n, counter);
        }    
       	else
       		return true;
    }
    
    
    public static boolean isLucky(int n)
    {
        return check(n,2);
    }
    
}